src/Yesod/Paginator/Paginate.hs

```diff
@@ -86,7 +121,21 @@ selectPaginated' per filters options p =
         )
 
 getCurrentPage :: MonadHandler m => m PageNumber
-getCurrentPage = fromMaybe 1 . go <$> lookupGetParam "p"
+getCurrentPage =
+    getCurrentPageWith (paginationConfigPageParamName defaultPaginationConfig)
+
+getCurrentPageWith :: MonadHandler m => PageParamName -> m PageNumber
+getCurrentPageWith pageParamName = fromMaybe 1 . go <$> lookupGetParam
+    (unPageParamName pageParamName)
   where
     go :: Maybe Text -> Maybe PageNumber
     go mp = readIntegral . unpack =<< mp
+
+newtype PageParamName = PageParamName { unPageParamName :: Text }
+    deriving (Eq)
+    deriving newtype (Read, Show, PathPiece)
+
+data PaginationConfig = PaginationConfig
+  { paginationConfigPageParamName :: PageParamName
+  , paginationConfigPerPage :: PerPage
+  }
```

Comment:
Not sure if these types should live in another module?

Reply:
I tend to make a lot of modules, usually focused around (and named after) a single data type (such as `PaginationConfig`) and related functions. So I would do that here. That said, this is one of my older projects, so I'm not sure if it's following that guide generally or not. So, if for example, the fact that some other data type is not in its own leaf module make sit harder to do it for this one, don't worry about it.

Reply:
Ok, sounds good. I moved these parts to a separate module.